### ISSUE TITLE
`ExitGracefully` is not included in enough source files

### ISSUE THREAD

**Issue by Zeitsperre** [CSHS-CWRA/RavenHydroFramework#24]
### Setup Information

- Raven version: 3.8
- Operating System: Linux

### Description

The latest tag (v3.8) appears to be missing some include statements for the `ExitGracefully` function. This is causing a bunch of compiling errors when building with `cmake`.

### Steps To Reproduce

```
$ git clone git@github.com:Ouranosinc/raven-hydro.git
$ git checkout from-offcial-git
$ pip install --user --editable . --verbose
```

### Additional context

```
...
      /usr/bin/ld: CMakeFiles/raven.dir/src/RavenHydroFramework/src/Abstraction.cpp.o: in function `CmvAbstraction::GetParticipatingParamList(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, class_type*, int&) const':
      Abstraction.cpp:(.text+0xbe6): undefined reference to `ExitGracefully(char const*, exitcode)'
      /usr/bin/ld: CMakeFiles/raven.dir/src/RavenHydroFramework/src/Advection.cpp.o: in function `CmvAdvection::GetRatesOfChange(double const*, CHydroUnit const*, optStruct const&, time_struct const&, double*) const':
      Advection.cpp:(.text+0x6c1): undefined reference to `ExitGracefully(char const*, exitcode)'
      /usr/bin/ld: CMakeFiles/raven.dir/src/RavenHydroFramework/src/Advection.cpp.o: in function `CmvAdvection::Initialize()':
      Advection.cpp:(.text+0xcbc): undefined reference to `ExitGracefully(char const*, exitcode)'
      /usr/bin/ld: CMakeFiles/raven.dir/src/RavenHydroFramework/src/Assimilate.cpp.o: in function `CModel::InitializeDataAssimilation(optStruct const&)':
      Assimilate.cpp:(.text+0x42b): undefined reference to `ExitGracefully(char const*, exitcode)'
      /usr/bin/ld: Assimilate.cpp:(.text+0x443): undefined reference to `ExitGracefully(char const*, exitcode)'
      /usr/bin/ld: CMakeFiles/raven.dir/src/RavenHydroFramework/src/Baseflow.cpp.o:Baseflow.cpp:(.text+0x818): more undefined references to `ExitGracefully(char const*, exitcode)' follow
      collect2: error: ld returned 1 exit status
      ninja: build stopped: subcommand failed.
```

### Contribution

- [X] I would be willing/able to open a Pull Request to address this bug.

**Comment by Zeitsperre** [comment]
See: https://github.com/Ouranosinc/raven-hydro/pull/24 for work in progress

**Comment by huard** [comment]
The error can be reproduced by compiling Raven with the `-DCMAKE_BUILD_TYPE=Release` flag.

**Comment by Zeitsperre** [comment]
This appears to have been fixed at some point. We now have build checks on all pushes to ensure that we know which changes are breaking the builds (debug and release).

@analytophile Can we get a new version tagged so that the Python supply chain can benefit? If we aren't ready for a patch release, something like v3.8.0.xxx would be fine.

**Comment by analytophile** [comment]
This issues has been resolved, and a new tagged version v3.8.1 online.
